src/V3Randomize.cpp

```diff
@@ -438,13 +456,97 @@ class ConstraintExprVisitor final : public VNVisitor {
     }
 };
 
-template <typename TreeNodeType>
-class CaptureFrame final {
-    TreeNodeType* m_treep;  // Original tree
+class ClassLookupHelper final {
+    const std::set<AstNodeModule*> m_visibleModules;
+    const std::vector<AstNodeModule*> m_symLookupOrder;
+    std::map<AstNode*, AstNodeModule*> m_classMap;
+
+    using inits_t = std::tuple<std::set<AstNodeModule*>, std::vector<AstNodeModule*>>;
+
+    // BFS search
+    template <typename Action>
+    static void foreachSuperClass(AstClass* classp, Action action) {
+        std::queue<AstClass*> classes;
+        classes.push(classp);
+        while (!classes.empty()) {
+            classp = classes.front();
+            classes.pop();
+            for (AstClassExtends* extendsp = classp->extendsp(); extendsp;
+                 extendsp = VN_AS(extendsp->nextp(), ClassExtends)) {
+                AstClass* const superClassp
+                    = VN_AS(extendsp->childDTypep(), ClassRefDType)->classp();
+                action(superClassp);
+                classes.push(superClassp);
+            }
+        }
+    }
+
+    static inits_t init(AstClass* classp) {
```

Comment:
```suggestion
    static ClassInits init(AstClass* classp) {
```

Reply:
I've removed the `m_symLookupOrder` field, because it was no longer used, which simplified the code and made this part unnecessary, so it's gone now.